dockstore-integration-testing/src/test/java/io/dockstore/client/cli/LimitedCRUDClientIT.java

```diff
@@ -138,12 +143,79 @@ public void testToolCreation(){
         api.createHostedTool("awesomeTool" + 10, Registry.QUAY_IO.toString().toLowerCase(), DescriptorLanguage.CWL_STRING, "coolNamespace", null);
     }
 
+    @Test
+    public void testOverrideEntryLimit() {
+        ApiClient webClient = BaseIT.getWebClient(BaseIT.ADMIN_USERNAME);
+        HostedApi api = new HostedApi(webClient);
+
+        // Change limits for current user
+        UsersApi usersApi = new UsersApi(webClient);
+        User user = usersApi.getUser();
+        Limits limits = new Limits();
+        limits.setHostedEntryCountLimit(NEW_LIMITS);
+        usersApi.setUserLimits(user.getId(), limits);
+        DockstoreTool hostedTool = api.createHostedTool("awesomeTool", Registry.QUAY_IO.toString().toLowerCase(), DescriptorLanguage.CWL_STRING, "coolNamespace", null);
+        assertNotNull("tool was not created properly", hostedTool);
+        // createHostedTool() endpoint is safe to have user profiles because that profile is your own
+        assertEquals("One user should belong to this tool, yourself",1, hostedTool.getUsers().size());
+
+        // test repeated workflow creation up to limit
+        for(int i = 1; i <= NEW_LIMITS - 1; i++) {
+            api.createHostedTool("awesomeTool" + i, Registry.QUAY_IO.toString().toLowerCase(), DescriptorLanguage.CWL_STRING, "coolNamespace", null);
+        }
+
+        thrown.expect(ApiException.class);
+        api.createHostedTool("awesomeTool" + NEW_LIMITS, Registry.QUAY_IO.toString().toLowerCase(), DescriptorLanguage.CWL_STRING, "coolNamespace", null);
+    }
+
     @Test
     public void testToolVersionCreation() throws IOException {
         ApiClient webClient = BaseIT.getWebClient(BaseIT.ADMIN_USERNAME);
         HostedApi api = new HostedApi(webClient);
         DockstoreTool hostedTool = api.createHostedTool("awesomeTool", Registry.QUAY_IO.toString().toLowerCase(), DescriptorLanguage.CWL_STRING, "coolNamespace", null);
 
+        List<SourceFile> sourceFiles = generateSourceFiles();
+
+        api.editHostedTool(hostedTool.getId(), sourceFiles);
+
+        // test repeated workflow version creation up to limit
+        for(int i = 1; i <= 9; i++) {
+            //TODO: this is kind of dumb, we should check for no-change versions as a hotfix
```

Comment:
yup, please add an issue to the 1.6.X milestone (you'll probably need to create it)